pkg/operator2/configobservation/apiserver/observe_cors.go

```diff
@@ -0,0 +1,62 @@
+package apiserver
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var clusterDefaultCORSAllowedOrigins = []string{
+	`//127\.0\.0\.1(:|$)`,
+	`//localhost(:|$)`,
+}
+
+func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
will we move these to apiserver-library-go ?

Reply:
yes, I plan on doing that, we have reuse of that in the OAS-o and here now

Reply:
added it to lib-go with some unit tests as well: https://github.com/openshift/library-go/pull/668